sdk/python/test_data/pipelines/pipeline_with_google_artifact_type.py

```diff
@@ -38,7 +40,7 @@ def __init__(self, name: str, uri: str, metadata: dict) -> None:
         def path(self) -> str:
             return self.uri.replace('gs://', '/')
 
-    class VertexDataset:
+    class VertexDataset(dsl.Artifact):
```

Comment:
Would `dsl.Dataset` make more sense here?

Reply:
I think that might falsely suggest some form of artifact polymorphism. My preference is to inherit from the base class which contains the minimum set of functionality (`.path`, constructor) that we're interested in inheriting, without suggesting that it gets "dataset" properties.

Reply:
That's fine. We can revisit this later. IIRC, `dsl.Dataset` doesn't have any special properties other than being rendered with a different icon in Vertex. It's likely out of scope here, but we should sort out a clean path for our artifact usage so that we would have certain answer/guidline to questions like this.